server/datastore/mysql/hosts.go

```diff
@@ -717,6 +717,43 @@ func shouldCleanTeamPolicies(currentTeamID, newTeamID *uint) bool {
 	return *currentTeamID != *newTeamID
 }
 
+func (ds *Datastore) EnrollOrbit(ctx context.Context, hardwareUUID string, orbitNodeKey string) (*fleet.Host, error) {
+	if orbitNodeKey == "" {
+		return nil, ctxerr.New(ctx, "orbit node key is empty")
+	}
+
+	if hardwareUUID == "" {
+		return nil, ctxerr.New(ctx, "harware uuid is empty")
+	}
+
+	var host fleet.Host
+	err := ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		err := sqlx.GetContext(ctx, tx, &host, `SELECT id FROM hosts WHERE uuid = ?`, hardwareUUID)
+		switch {
+		case err != nil && !errors.Is(err, sql.ErrNoRows):
+			return ctxerr.Wrap(ctx, err, "orbit enroll error selecting host details")
+		case errors.Is(err, sql.ErrNoRows):
+			sqlInsert := `INSERT INTO hosts (osquery_host_id, uuid, orbit_node_key) VALUES (?, ?, ?)`
+			_, err := tx.ExecContext(ctx, sqlInsert, hardwareUUID, hardwareUUID, orbitNodeKey)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "orbit enroll error inserting host details")
+			}
+		default:
+			sqlUpdate := `UPDATE hosts SET orbit_node_key = ? WHERE uuid = ? `
+			_, err := tx.ExecContext(ctx, sqlUpdate, orbitNodeKey, hardwareUUID)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "orbit enroll error updating host details")
+			}
+
+		}
```

Comment:
A way to rewrite so that `errors.Is(err, sql.ErrNoRows)` is only checked in one `case`:
```go
switch {
case err == nil:
  // ...
case errors.Is(err, sql.ErrNoRows):
  // ...
default: // err != nil && !errors.Is(err, sql.ErrNoRows)
}
```